cpp/src/copying/contiguous_split.cu

```diff
@@ -1158,122 +1268,721 @@ std::vector<packed_table> contiguous_split(cudf::table_view const& input,
+      0, out_to_in_index_function{chunk_iter_state->d_batch_offsets.begin(), (int)num_bufs});
+
+    auto values = thrust::make_transform_iterator(
+      chunk_iter_state->d_batched_dst_buf_info.begin(),
+      [] __device__(dst_buf_info const& info) { return info.valid_count; });
+
+    thrust::reduce_by_key(rmm::exec_policy(stream, mr),
+                          keys,
+                          keys + num_batches_total,
+                          values,
+                          thrust::make_discard_iterator(),
+                          dst_valid_count_output_iterator{d_orig_dst_buf_info});
+
+    CUDF_CUDA_TRY(cudaMemcpyAsync(h_orig_dst_buf_info,
+                                  d_orig_dst_buf_info,
+                                  partition_buf_size_and_dst_buf_info->dst_buf_info_size,
+                                  cudaMemcpyDefault,
+                                  stream.value()));
+
+    stream.synchronize();
+
+    // not necessary for the non-chunked case, but it makes it so further calls to has_next
+    // return false, just in case
+    chunk_iter_state->advance_iteration();
+
+    return make_packed_tables();
+  }
+
+  cudf::size_type contiguous_split_chunk(cudf::device_span<uint8_t> const& user_buffer)
+  {
+    CUDF_FUNC_RANGE()
+    CUDF_EXPECTS(
+      user_buffer.size() == user_buffer_size,
+      "Cannot use a device span smaller than the output buffer size configured at instantiation!");
+    CUDF_EXPECTS(has_next(), "Cannot call contiguous_split_chunk with has_next() == false!");
+
+    auto [starting_batch, num_batches_to_copy] =
+      chunk_iter_state->get_current_starting_index_and_buff_count();
+
+    // perform the copy.
+    copy_data(num_batches_to_copy,
+              starting_batch,
+              src_and_dst_pointers->d_src_bufs,
+              src_and_dst_pointers->d_dst_bufs,
+              chunk_iter_state->d_batched_dst_buf_info,
+              user_buffer.data(),
+              stream);
+
+    // We do not need to post-process null counts since the null count info is
+    // taken from the source table in the contiguous_split_chunk case (no splits)
+    return chunk_iter_state->advance_iteration();
+  }
+
+  std::vector<packed_table> make_empty_packed_table()
+  {
+    // sanitize the inputs (to handle corner cases like sliced tables)
+    std::vector<std::unique_ptr<column>> empty_columns;
+    empty_columns.reserve(input.num_columns());
+    std::transform(
+      input.begin(), input.end(), std::back_inserter(empty_columns), [](column_view const& col) {
+        return cudf::empty_like(col);
+      });
+    std::vector<cudf::column_view> empty_column_views;
+    empty_column_views.reserve(input.num_columns());
+    std::transform(empty_columns.begin(),
+                   empty_columns.end(),
+                   std::back_inserter(empty_column_views),
+                   [](std::unique_ptr<column> const& col) { return col->view(); });
```

Comment:
I don't think you need the vector of `column` objects if they are all empty -- no data, no bitmask.
```suggestion
    std::vector<cudf::column_view> empty_column_views;
    empty_column_views.reserve(input.num_columns());
    std::transform(input.begin(), input.end(),
                   std::back_inserter(empty_column_views),
                   [](column_view const& col) { return cudf::empty_like(col)->view(); });
```

Reply:
Yay that is better.

Reply:
This works, and tests pass (we check the empty case). Just to confirm: the `column` that a `column_view` has to be in scope iff it were to have GPU data (validity, data), otherwise it's fine as the host-side functions are guaranteed to work.

Reply:
Patch: https://github.com/rapidsai/cudf/pull/13260/commits/8c344a4beb320d37ad1c69cd9af9beb9f699d404

Reply:
That is correct. The `column` object returned from `empty_like` is destroyed after the `return` statement but the resulting `column_view` contains only copied data from the `column` in this special case.